src/Aspire.Hosting/ApplicationModel/ResourceCommandAnnotation.cs

```diff
@@ -9,7 +9,7 @@ namespace Aspire.Hosting.ApplicationModel;
 /// Represents a command annotation for a resource.
 /// </summary>
 [DebuggerDisplay("Type = {GetType().Name,nq}, Name = {Name}")]
-public sealed class ResourceCommandAnnotation : IResourceAnnotation
+public sealed class ResourceCommandAnnotation : ResourceCommandAnnotationBase
```

Comment:
We shouldn't need to do this. The annotation provides a callback that should be able to capture everything it needs in the closure.

Reply:
We have a different API here - we dont want any callback be provided by the user, so it should be a different annotation. I can avoid using a base type here, but callback is definitely not suitable here